src/mixins/Syncable.ts

```diff
@@ -0,0 +1,170 @@
+import { Instance, types } from 'mobx-state-tree';
+
+/**
+ * Supress all additional events during this window in ms.
+ * 100ms is too short to notice, but covers enough frames (~6) for back and forth events.
+ */
+export const SYNC_WINDOW = 100;
+
+export type SyncEvent = 'play' | 'pause' | 'seek' | 'speed';
+
+/**
+ * Currently only for reference, MST mixins don't allow to apply this interface
+ */
+export interface SyncTarget {
+  name: string;
+  sync: string;
+  syncSend(data: SyncData, event: SyncEvent): void;
+  syncReceive(data: SyncData, event: SyncEvent): void;
+  registerSyncHandlers(): void;
+  destroy(): void;
+}
+
+export interface SyncDataFull {
+  time: number;
+  playing: boolean;
+  speed: number;
+}
+
+export type SyncData = Partial<SyncDataFull>;
+
+/**
+ * Sync group of tags with each other; every tag should be registered
+ */
+export class SyncManager {
+  syncTargets = new Map<string, Instance<typeof SyncableMixin>>();
+  locked: string | null = null; // refers to the main tag, which locked this sync
+
+  register(syncTarget: Instance<typeof SyncableMixin>) {
+    this.syncTargets.set(syncTarget.name, syncTarget);
+  }
+
+  unregister(syncTarget: Instance<typeof SyncableMixin>) {
+    this.syncTargets.delete(syncTarget.name);
+    // @todo remove manager on empty set
+  }
+
+  /**
+   * Sync `origin` state (in `data`) to connected tags.
+   * No back-sync to origin of the event.
+   * During SYNC_WINDOW only events from origin are processed, others are skipped
+   * @param {SyncData} data state to sync between connected tags
+   * @param {string} event name of event, supplementary info, actions should rely on data
+   * @param {string} origin name of the tag triggered event
+   * @returns {boolean} false if event was suppressed, because it's inside other event sync window
+   */
+  sync(data: SyncData, event: SyncEvent, origin: string) {
+    // @todo remove
+    if (!this.locked || this.locked === origin) console.log('SYNC', { event, locked: this.locked, data, origin });
```

Comment:
Perhaps it's not a bad idea to highlight this with this comment. I'm not sure when we plan to be remove it.